### ISSUE TITLE
Support refresh in ES 5.y.z

### ISSUE THREAD

**Issue by dkoston** [couchbaselabs/couchbase-elasticsearch-connector#154]
https://www.elastic.co/guide/en/elasticsearch/reference/5.0/docs-refresh.html#docs-refresh

This is a nice new option in ES that allows you to customize how fast results appear in ElasticSearch queries. Please expose the ability to configure passing ?refresh=X when adding/updating documents to indexes

**Comment by dnault** [comment]
Setting `refresh` to `true` will put more load on the Elasticsearch server. Is that acceptable in your use case? Or were you hoping to go the other direction, defer indexing to reduce ES load?

**Comment by dnault** [comment]
After a bit more research and though, it looks like this probably isn't something that makes sense for the connector. As noted in the description of the `IMMEDIATE` refresh policy:

> This refresh policy does not scale for high indexing or search throughput
but is useful to present a consistent view to for indices with very low traffic.

If the goal is to minimize the time between when a document is written to Couchbase and when it is searchable in Elasticsearch, then forcing a refresh after every write does not help in the long run. Even though the *first* batch of documents might be searchable more quickly, subsequent batches actually take *longer* to appear in search results just because the forced refresh acts as a bottleneck that significantly delays future indexing.

If the goal is to have consistent search results (that is, to write a document in Couchbase and then make an Elasticsearch query that returns the document), forcing a refresh doesn't solve that problem either. The time between when a document is written to Couchbase and when the document reaches the connector to be indexed can vary based on the current replication load. The best guarantee the connector can offer is that the document will appear in Elasticsearch "eventually." Making it visible in search results a little bit sooner ("eventually - 1") doesn't offer stronger consistency.

tldr; Forcing a refresh is the solution to a different problem.

**Comment by dkoston** [comment]
Sorry, I should have closed this one. We had some record discrepancies where XDCR didn’t have the current records in couchbase yet on UPDATE. We built a system that kept track of the changes made in the UPDATES and merged ES results with up to date records from couchbase.
